extensions/amp-audioburst-audio/0.1/amp-audioburst-audio.js

```diff
@@ -0,0 +1,460 @@
+    setStyles(sliderL2, {
+      'position': 'absolute',
+      'background': '#ae21ff',
+      'border': '2px solid #fff',
+      'border-radius': '50%',
+      'width': '1em',
+      'height': '1em',
+      'top': '-0.57em',
+      'left': '-0.65em',
+      'box-shadow': '3px 1px 8px 0px rgba(18, 12, 48, 0.2)',
+      'z-index': '2',
+      'transition': '0.3s linear left',
+    });
+
+    slider.appendChild(sliderL1);
+    slider.appendChild(sliderL2);
+
+    const setAudioPosition = x => {
+      const rectangle = this.viewport_.getLayoutRect(slider);
+      const eventX = x ? x : 0;
+      if (rectangle && this.audio_) {
+        const percent = (eventX - rectangle.x) / rectangle.width;
+        this.audio_.currentTime = this.duration * percent;
+      }
+    };
+
+    listen(slider, 'touchstart',
+        event => setAudioPosition(event.changedTouches[0].clientX));
+    listen(slider, 'click', event => setAudioPosition(event.x));
+
+    let sliderDraging = false;
+    listen(sliderL2, 'touchstart', () => sliderDraging = true);
+    listen(sliderL2, 'mousedown', () => sliderDraging = true);
+    listen(sliderL2, 'touchmove', event => {
+      if (sliderDraging) {
+        setAudioPosition(event.changedTouches[0].clientX);
+      }
+    });
+    listen(sliderL2, 'mousemove', event => {
+      if (sliderDraging) {
+        setAudioPosition(event.x);
+      }
+    });
+    listen(sliderL2, 'touchend', event => {
+      if (sliderDraging) {
+        setAudioPosition(event.changedTouches[0].clientX);
+      }
+      sliderDraging = false;
+    });
+    listen(sliderL2, 'mouseup', event => {
+      if (sliderDraging) {
+        setAudioPosition(event.x);
+      }
+      sliderDraging = false;
+    });
+    this.slider_ = slider;
+  }
+
+  /** @private */
+  updateSlider_() {
+    if (this.slider_) {
+      const l1 = this.slider_.querySelector('#ampAudioburstAudioSliderL1');
+      const l2 = this.slider_.querySelector('#ampAudioburstAudioSliderL2');
+      const width = this.currentTime * 100 / this.duration;
+      if (l1 && this.currentTime >= 0 && this.duration > 0) {
+        setStyle(l1, 'width', width + '%');
+      }
+      if (l2 && this.currentTime >= 0 && this.duration > 0 && width <= 100) {
+        setStyle(l2, 'left', `calc(${width}% - 0.65em)`);
+      }
+    }
+  }
+
+  /** @private
+   * @param {number} seconds
+  */
+  convertTime_(seconds) {
+    if (seconds == undefined || seconds === 0) {
+      return '0:00';
+    }
+    let result = '';
+    seconds = Math.floor(seconds);
+    if (seconds < 60) {
+      result =
+        `0:${(seconds < 10 ? '0' + seconds.toString() : seconds.toString())}`;
+    } else {
+      const minutes = Math.floor(seconds / 60);
+      seconds = seconds % 60;
+      result = `${minutes.toString()}:`;
+      result +=
+        `${(seconds < 10 ? '0' + seconds.toString() : seconds.toString())}`;
+    }
+    return result;
+  }
+}
+
+AMP.registerElement('amp-audioburst-audio', AmpAudioburstAudio);
```

Comment:
Following what other extensions do, this needs to be

```js
const TAG = 'amp-audioburst-audio';

AMP.extension(TAG, '0.1', AMP => {
  AMP.registerElement(TAG, AmpAudioburstAudio);
});
```

Reply:
done